service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -120,7 +120,7 @@ public class EhrExtractStatusService {
     private static final String ACKS_SET = ACK_HISTORY + DOT + ACKS;
     public static final int EHR_EXTRACT_SENT_DAYS_LIMIT = 8;
     private static final String UNEXPECTED_CONDITION_ERROR_CODE = "99";
-    public static final String UNEXPECTED_CONDITION_ERROR_MESSAGE = "The acknowledgement has been received after 8 days";
+    public static final String UNEXPECTED_CONDITION_ERROR_MESSAGE = "No acknowledgement has been received within 8 days";
```

Comment:
Your functionality as it currently stands is tightly coupled to the wording of this string. You store a copy of the string in the MongoDB, and then use it to check the record matches again.

The side effect of which is that if we ever want to change this string, then we have to be aware of any existing records in the DB, and updating them to match. Otherwise the check for `hasEhrStatusReceivedAckWithUnexpectedConditionErrors` will return `false`.